File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        results = {}
        
        for index,num in enumerate(nums):
            potential = target - num
            if potential in results:
                return [results[potential], index]
            else:
                results[num] = index
        

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int num = 0; num < nums.length; num++) {
            int potentialMatch = target - nums[num];
            if (map.containsKey(potentialMatch)) {
                return new int[] {map.get(potentialMatch), num};
            }
            map.put(nums[num], num);
        }
        return null;
    }
}